### DISCUSSION TITLE
Getting "Unexpected server error: 200" on  running the Websocket connect test.

### DISCUSSION THREAD

**Discussion by Vikrant87-lego** [artilleryio/artillery#1161] [Artillery]
Hi ,

I am facing an issue where i get below mentioned error:

"Started phase 0, duration: 2s @ 10:43:30(+0200) 2021-08-31
 ..   ws Error: Unexpected server response: 200
  ws     at ClientRequest.<anonymous> (/Users/username/.nvm/versions/node/v14.17.1/lib/node_modules/artillery/node_modules/ws/lib/websocket.js:542:5)
  ws     at ClientRequest.emit (events.js:375:28)
  ws     at ClientRequest.emit (domain.js:470:12)
  ws     at HTTPParser.parserOnIncomingClient [as onIncoming] (_http_client.js:647:27)
  ws     at HTTPParser.parserOnHeadersComplete (_http_common.js:126:17)
  ws     at TLSSocket.socketOnData (_http_client.js:515:22)
  ws     at TLSSocket.emit (events.js:375:28)
  ws     at TLSSocket.emit (domain.js:470:12)
  ws     at addChunk (internal/streams/readable.js:290:12)
  ws     at readableAddChunk (internal/streams/readable.js:265:9) +0ms"

Script:
config:
  target: "wss://baseURL"
  phases:
    - duration: 2
      arrivalRate: 2
  variables:
    token: "Bearer (Token)"

  ws:
    headers:
      authorization: {{ token }}

scenarios:
  - name: "Connecting to Websocket"
    engine: ws
    flow:
      - connect: "/Websocket/"

Can someone please help me understand what is wrong with the script?

**Comment by flentini** [comment]
hey @Vikrant87-lego 

`ws Error: Unexpected server response: 200`

your server is not handling the ws handshake correctly, a websocket server should respond with 101, not 200. You could try setting your endpoint in the `connect` step as:

`- connect: "{{target}}/Websocket"`

you can run artillery prefixing it with `DEBUG=ws` and it will print out the full url it's trying to connect to: is it what you are expecting ?
